lib/ts/bwprint.cc

```diff
@@ -0,0 +1,410 @@
+          _type = *sz;
+          if (!++sz)
+            return;
+        }
+      }
+    }
+  }
+}
+
+/** This performs generic alignment operations.
+
+    If a formatter specialization performs this operation instead, that should result in output that
+    is at least @a spec._min characters wide, which will cause this function to make no further
+    adjustments.
+ */
+void
+ts::detail::bwf_aligner(BWFSpec const &spec, BufferWriter &w, BufferWriter &lw)
+{
+  size_t size = lw.size();
+  size_t min  = spec._min;
+  if (size < min) {
+    size_t delta = min - size; // note - size <= extent -> size < min
+    switch (spec._align) {
+    case BWFSpec::Align::NONE: // same as LEFT for output.
+    // fall through
+    case BWFSpec::Align::LEFT:
+      w.fill(size);
+      while (delta--)
+        w.write(spec._fill);
+      size = 0; // cancel additional fill.
+      break;
+    case BWFSpec::Align::RIGHT:
+      std::memmove(w.auxBuffer() + delta, w.auxBuffer(), size);
+      while (delta--)
+        w.write(spec._fill);
+      break;
+    case BWFSpec::Align::CENTER:
+      if (delta > 1) {
+        size_t d2 = delta / 2;
+        std::memmove(w.auxBuffer() + (delta / 2), w.auxBuffer(), size);
+        while (d2--)
+          w.write(spec._fill);
+      }
+      w.fill(size);
+      delta = (delta + 1) / 2;
+      while (delta--)
+        w.write(spec._fill);
+      size = 0; // cancel additional fill.
+      break;
+    case BWFSpec::Align::SIGN:
+      break;
+    }
+  }
+  w.fill(size);
+}
+
+// Numeric conversions
+namespace ts::detail
+{
+// Conversions from remainder to character, in upper and lower case versions.
+namespace
+{
+  char UPPER_DIGITS[] = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+  char LOWER_DIGITS[] = "0123456789abcdefghijklmnopqrstuvwxyz";
+}
+/// Templated radix based conversions. Only a small number of radix are supported
+/// and providing a template minimizes cut and paste code while also enabling
+/// compiler optimizations (e.g. for power of 2 radix the modulo / divide become
+/// bit operations).
+template <size_t RADIX>
+size_t
+bwf_to_radix(uintmax_t n, char *buff, size_t width, char *digits)
+{
+  static_assert(1 < RADIX && RADIX <= 36);
+  char *out = buff + width;
+  if (n) {
+    while (n) {
+      *--out = digits[n % RADIX];
+      n /= RADIX;
+    }
+  } else {
+    *--out = '0';
+  }
+  return (buff + width) - out;
+}
+
+BufferWriter &
+bwf_integral_formatter(BufferWriter &w, BWFSpec const &spec, uintmax_t i, bool neg_p)
+{
+  size_t n  = 0;
+  int width = static_cast<int>(spec._min); // amount left to fill.
+  string_view prefix;
+  string_view neg;
+  char buff[std::numeric_limits<uintmax_t>::digits + 1];
+
+  if (neg_p)
+    neg = "-"_sv;
+  else if (spec._sign != '-')
+    neg = string_view{&spec._sign, 1};
+  switch (spec._type) {
+  case 'x':
+    if (spec._radix_lead_p)
+      prefix = "0x"_sv;
+    n        = detail::bwf_to_radix<16>(i, buff, sizeof(buff), detail::LOWER_DIGITS);
```

Comment:
Putting braces around the body of the `if` will fix this weird formatting and also match our style guide.

Reply:
Haha, yeah, that's some serious ugly there, isn't it?